src/main/java/com/google/songgame/servlets/GameServlet.java

```diff
@@ -159,13 +167,47 @@ public static YouTube getService() throws GeneralSecurityException, IOException
   }
 
   /**
-   * Returns video ID
+   * Retrieve random video from array of videos and stores video in datastore
    */
   private String getRandomVideo(ArrayList<String> playlistVideos) {
     Random randomGenerator = new Random();
     int playlistSize = playlistVideos.size();
     int index = randomGenerator.nextInt(playlistSize);
     String videoId = playlistVideos.get(index);
+    
+    try {
+      Video currentVideo = getVideoInfo(videoId);
+      setVideoInfo(currentVideo);
+    } catch (Exception e) {
+      System.err.println("ERROR: Could not read video");
+    }
+
     return videoId;
   }
+
+  /**
+   * Retrieves video information given a particular video ID
+   */
+  private Video getVideoInfo(String videoId) throws GeneralSecurityException, IOException, GoogleJsonResponseException {
+    YouTube youtubeService = getService();
+    YouTube.Videos.List request = youtubeService.videos().list("snippet");
+    VideoListResponse response = request.setId(videoId).execute();
+    Video video = response.getItems().get(0);
+    return video;    
+  }
+
+  /**
+   * Set video information in Datastore
+   */
+  private void setVideoInfo(Video video) {
+    String videoTitle = video.getSnippet().getTitle();
+    long timestamp = System.currentTimeMillis();
+    
+    Entity videoEntity = new Entity("Video");
+    videoEntity.setProperty("title", videoTitle);
+    videoEntity.setProperty("timestamp", timestamp);
```

Comment:
The name "timestamp" is pretty generic and it's not clear what code that reads videos from the database is supposed to do with this property. This would especially become apparent if we needed to add another time-related property to this entity, since it would be hard to distinguish between the two unless we give them descriptive names.

It seems this is the time the video was first fetched from the playlist, so maybe "fetchTime" will do, or even "creationTime", though that could be confused with the actual time the video was created in youtube. Let me know what you think about this or if you have other suggestions for how to name this.